eZ/Publish/Core/Persistence/Legacy/Content/Language/MaskGenerator.php

```diff
@@ -36,39 +36,26 @@ public function __construct(LanguageHandler $languageHandler)
     /**
      * Generates a language mask from the keys of $languages.
      *
+     * @deprecated Move towards using {@see generateLanguageMaskFromLanguageMap()} or the other specific methods.
```

Comment:
TBH instead of introducing `generateLanguageMaskFromLanguageMap` I'd refactor this method and deprecate it suggesting to use `generateLanguageMaskFromLanguageCodes(array $languageCodes, bool $isAlwaysAvailable = false): int` instead.

The architecture where the input array was e.g.
```
[
    'always-available' => 1,
    'eng-GB' => 'value',
    'nor-NO' => 'value',
]
```
was not the greatest one.

Reply:
could do that, but if I remember right the format is used a few places, so if I get rid of the method we'll need proxy methods added around to handle conversion.

But as a middle ground I can deprecate as you say, buyt change them back to use `generateLanguageMask` to slim down diff. Letting it be something we cleanup in the future.

Reply:
> could do that, but if I remember right the format is used a few places, so if I get rid of the method we'll need private helper methods added around to handle conversion.
> 
> But as a middle ground I can deprecate as you say, but just change those with such format still back to use `generateLanguageMask` to slim down diff. Letting it be something we aim to cleanup in the future.

Hmm, but AFAICS `generateLanguageMaskFromLanguageMap` is used by `generateLanguageMask` only. What else would need to be changed back?

Reply:
Indeed you are right, after the refactorings here and the other PRs it's not in use anymore.
Fixing